stac_fastapi/core/stac_fastapi/core/serializers.py

```diff
@@ -121,6 +122,11 @@ def db_to_stac(cls, item: dict, base_url: str) -> stac_types.Item:
             assets=assets,
         )
 
+        private_data_field = os.getenv("PRIVATE_DATA_FIELD")
+        if get_bool_env("HIDE_PRIVATE_DATA") and private_data_field:
+            stac_item.pop(private_data_field, None)
+        return stac_item
```

Comment:
1. let's allow for a list of private data fields. `PRIVATE_DATA_FIELDS`
2. most Users will have private fields listed under `properties` so we need to account for nested fields